test/adapter/json_api/errors_test.rb

```diff
@@ -0,0 +1,116 @@
+
+- id: a unique identifier for this particular occurrence of the problem.
+- links: a links object containing the following members:
+  - about: a link that leads to further details about this particular occurrence of the problem.
+- status: the HTTP status code applicable to this problem, expressed as a string value.
+- code: an application-specific error code, expressed as a string value.
+- title: a short, human-readable summary of the problem that SHOULD NOT change from occurrence to occurrence of the problem, except for purposes of localization.
+- detail: a human-readable explanation specific to this occurrence of the problem.
+- source: an object containing references to the source of the error, optionally including any of the following members:
+  -  pointer: a JSON Pointer [RFC6901] to the associated entity in the request document [e.g. "/data" for a primary data object, or "/data/attributes/title" for a specific attribute].
+  -  parameter: a string indicating which query parameter caused the error.
+- meta: a meta object containing non-standard meta-information about the error.
+
+=end
+
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonApi
+        class ErrorsTest < Minitest::Test
+          include ActiveModel::Serializer::Lint::Tests
+
+          # see
+          # https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/errors.rb
+          # The below allows you to do:
+          #
+          #   model = ModelWithErrors.new
+          #   model.validate!            # => ["cannot be nil"]
+          #   model.errors.full_messages # => ["name cannot be nil"]
+          class ModelWithErrors
+            include ActiveModel::Serialization
+            def attributes
+              {
+                errors: errors,
+                name: name,
+              }
+            end
+
+            def id
+              object_id
+            end
+
+            def cache_key
+              "#{self.class.name.downcase}/#{id}-#{Time.now.utc.strftime("%Y%m%d%H%M%S%9N")}"
+            end
+
+            # Required dependency for ActiveModel::Errors
+            extend ActiveModel::Naming
+            def initialize
+              @errors = ActiveModel::Errors.new(self)
+            end
+            attr_accessor :name
+            attr_reader   :errors
+
+            # The following methods are needed to be minimally implemented
+
+            def read_attribute_for_validation(attr)
+              send(attr)
+            end
+
+            def self.human_attribute_name(attr, options = {})
+              attr
+            end
+
+            def self.lookup_ancestors
+              [self]
+            end
+          end
+
+          def setup
+            @resource = ModelWithErrors.new
+          end
+
+          def test_active_model_errors
+            options = {
+                serializer: ActiveModel::Serializer::ErrorSerializer,
+                adapter: :'json_api/error',
```

Comment:
until https://github.com/rails-api/active_model_serializers/pull/1017 is merged I can't use a plural adapter name since it calls 'classify'. womp womp